packages/jellyfish-api-core/__tests__/category/loan/createLoanScheme.test.ts

```diff
@@ -0,0 +1,106 @@
+    await container.stop()
+  })
+
+  it('should createLoanScheme', async () => {
+    const loanSchemeId = await client.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'scheme1' })
+    expect(typeof loanSchemeId).toStrictEqual('string')
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await container.generate(1)
+
+    const data = await container.call('listloanschemes')
+    const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+    expect(result.length).toStrictEqual(1)
+    expect(result[0]).toStrictEqual(
+      { id: 'scheme1', mincolratio: 200, interestrate: 2.5, default: false }
+    )
+  })
+
+  it('should not createLoanScheme if minConRatio is less than 100', async () => {
+    const promise = client.loan.createLoanScheme(99, new BigNumber(2.5), { id: 'scheme2' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nminimum collateral ratio cannot be less than 100\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if interestRate is less than 0.01', async () => {
+    const promise = client.loan.createLoanScheme(200, new BigNumber(0.00999), { id: 'scheme2' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\ninterest rate cannot be less than 0.01\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if same minConRatio and interestRate were created before', async () => {
+    const promise = client.loan.createLoanScheme(100, new BigNumber(1.5), { id: 'scheme2' }) // NOTE(jingyi2811): Failed because its minConRatio and interestRate are same as default
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme default with same interestrate and mincolratio already exists\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if same id was created before', async () => {
+    const promise = client.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'default' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme already exist with id default\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is an empty string', async () => {
+    const promise = client.loan.createLoanScheme(200, new BigNumber(2.5), { id: '' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is more than 8 chars long', async () => {
+    const promise = client.loan.createLoanScheme(200, new BigNumber(2.5), { id: '123456789' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
```

Comment:
Missing test for 
`Loan scheme %s with same interestrate and mincolratio pending on block %d` as seen in `src/masternodes/mn_checks.cpp:1944`

Reply:
Did some testing on this just now. During creation, `Loan scheme scheme1 with same interestrate and mincolratio already exists` is always thrown even though I create another scheme which has same number as other scheme which PENDING ON CERTAIN BLOCK.

It can only be triggered in UpdateLoanScheme rpc. I have added the test. See
https://github.com/DeFiCh/jellyfish/pull/530/files#diff-dd2673ebcdb99fa0f29b5daac47d8a4b45c4c4a49b90078b58b08d76daba84ffR157

Reply:
Ok looks good that way, as long as we test for `pending on block` somewhere